js/ui/date_box/ui.date_box.mask.js

```diff
@@ -96,7 +96,7 @@ const DateBoxMask = DateBoxBase.inherit({
     },
 
     _shouldUseOriginalHandler(e) {
-        return !this._useMaskBehavior() || this.option("opened") || (e && e.altKey);
+        return !this._useMaskBehavior() || this.option("opened") && e.key !== "Delete" && e.key !== "Backspace" || (e && e.altKey);
```

Comment:
The `e.key` value may vary depending on the browser.
Use `key = normalizeKeyName(e)` instead of it